### DISCUSSION TITLE
Thermostat testing

### DISCUSSION THREAD

**Discussion by chase9** [node-alarm-dot-com/homebridge-node-alarm-dot-com#112] [General]
This discussion thread is for organizing testing of the new thermostat support.

**Comment by chase9** [comment]
I'm running into an issue where my thermostat is still showing up in UI-X as C even though I use F. Does anyone else see this appearing incorrectly?

**Reply by pb30** [reply]
Just making sure, is your UI setting set to F? Shows fine in Home.app?

**Reply by chase9** [reply]
Everything seems fine in both Home assistant and in Home.app. The issue is limited to UI-X, but I'm not sure how many people are using UI-X as their frontend to HomeKit?

**Reply by chase9** [reply]
Nevermind, it looks like there is a separate spot in UI-X to set F/C. Changing this to F looks correct.

**Comment by chase9** [comment]
As far as handling thermostats with an automatic mode, I think for setting the thermostat state we should look to see if the desired temp is hotter or colder than the current temp and set to the appropriate heating or cooling mode. This is backed up by the homekit spec which has an automatic emum for target state but not for current state.

This would be a bit harder for if someone sets auto in the alarm.com portal and we need to handle it in the app. I suppose we would want to do the same by comparing the target temp to the current temp?
